datasets/covid19_tracking/state_screenshots/custom/web_scrape_and_generate_csv.py

```diff
@@ -82,7 +82,8 @@ def generate_csv_rows(url: str, gcs_path_prefix: str) -> typing.List[dict]:
         if td_date.text:
             current_date = datetime.strptime(td_date.text, "%B %d, %Y").date()
 
-        for link in td_screenshots.find_all("a"):
+        # Only get the last screenshot taken for the day
+        for link in td_screenshots.find_all("a")[-1:]:
```

Comment:
Is a for loop the right thing to use here? I guess we want to handle 0 or 1 elements?

Reply:
It'd also be nice to see a test updated for this logic.

Reply:
Based on the current state of the [screenshots website](http://screenshots.covidtracking.com/), we can always expect at least one `a` tag in that `td` element. Here's a screenshot of a successful pipeline run using that scraper logic across all the webpages in that site:

<img width="1381" alt="Screen Shot 2021-04-27 at 2 42 07 PM" src="https://user-images.githubusercontent.com/1208372/116295255-e2e39300-a766-11eb-86fe-0fdeec9ff691.png">

Also, this is a one-off pipeline to run, i.e. the C19 Tracking Project stopped updating the data last March 7. I guess we can keep it like that (one successful run for all time) to serve as an example pipeline that does web scraping?

Reply:
> It'd also be nice to see a test updated for this logic.

I agree. It's nice to have unit tests for the custom scripts. Those additional checks will minimize follow-up revisions for contributors in case something breaks that wasn't caught upstream.

But I'd like to have more time to think about the developer experience around testing custom scripts, i.e. where should contributors place them, any testing conventions to follow, updating the README to add that instruction, and so on. Hope you don't mind!

Reply:
Sounds good. Let's make an issue to track this (testing custom scripts) if you haven't already.

Reply:
Thanks! Got it tracked here: https://github.com/GoogleCloudPlatform/public-datasets-pipelines/issues/8